### ISSUE TITLE
Add keyboard navigation to this sim

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/friction#38]
We would like to add keyboard navigation to this sim, which means adding the interactive components to the focus order, and making the draggable components (book and magnified view) movable with arrow and WASD keys.

@mbarlow12 would you please work on this issue? We would like keyboard navigation in this sim to behave like the following:

- Focus order: "Magnified book" (molecules), zoomed out book, reset all button.
- Arrow keys/WASD keys move the draggable nodes similarly to the mouse.

The first step will be to move the drag handler in BASE to common code. Once that is done, the drag handler will be exercised here. @mbarlow12 I will let you know when  it is checked into scenery-phet.

**Comment by zepumph** [comment]
Here is a version of keyboard nav in friction with both dragable elements navigable by keyboard. I'm sure that there many problems and fine tunings to be done. @jessegreenberg and I just wanted to focus on the core implementation, and that seems to be working well, and our strategy looks promising. 

One issue I see is that if you are moving either object to the right, and then tab away, it gets stuck going to the right, and the behavior is very strange. I didn't see this in BASE, so I wonder if I'm just missing something.  

http://www.colorado.edu/physics/phet/dev/html/friction/1.4.0-dev.0/friction_en.html?accessibility

**Comment by zepumph** [comment]
@terracoda let me know what you think as an initial stab at it.

**Comment by jessegreenberg** [comment]
Looks great @zepumph! Removing my assignment on this issue.

**Comment by terracoda** [comment]
@zepumph, this is great! It seems to work fine for me. I do not see an issue when tabbing away.

**Regarding Tab Order:**
I think the tab order with the actual book first is good. Students first go to the real object and then zoom in, but we should check with the content expert. I'm not sure if order is important here. Not sure if cuing interaction with the chemistry book first is preferred?

**Regarding Focus Highlight (book)**
- Is is possible to draw the highlight more tightly around the book, so it goes along the edge of the book's cover? 
- The bottom part of the focus highlight obscures the tiny space between the books. Not sure what to do about that? My only thought would be to move the focus highlight inside the edge of the yellow book?

**Regarding Focus Highlight (zoomed in view)**
- It feels odd to me to just have the focus highlight around the particles as it make it feel like I am just moving the particles when I am actually moving the entire yellow block which includes the particles.
- Perhaps if the focus highlight included the space that the initial interaction cuing arrows take up as well as the particles, it would feel more like I was moving the book and the particles as one thing.

**Comment by terracoda** [comment]
@zepumph, in the case of BASE, we created a grab button that launches the custom balloon drag interaction. It will be interesting to see if we will need a two-step interaction here. The key is making the interaction clear in the non-visual scenario. We have less things going on here, so hopefully we won't need a two-step interaction.

Also, I  am wondering if the freezing upon tabbing away has anything to do with that difference in implementation? In BASE, there is an explicit grab and release interaction (a button), and then also a release on tabbing away. Here we would just have the "tab to" and "tab away", no explicit grab and release.

**Comment by zepumph** [comment]
@jessegreenberg I would love a little help on focus highlights here, I was thinking that I would just set my own focusHighlight, but when I do that I have to supply my own options to get my new Rectangle or whatever to look like the normal focushighlight. How do we go about fixing this. It would be great if we could have some options, maybe in FocusOverlay (not sure if that is the best) like stroke, line width and so on to make it easy to have to same style. 

Maybe there is even an easier way to accomplish this task, let's talk soon!

**Comment by zepumph** [comment]
In the mean time, I increased the top of the focus highlight to include where the arrows are on startup. 

@terracoda please comment on if you think this picture looks better (please disregard the styling of the focus highlight as much as possible)

![image](https://user-images.githubusercontent.com/6856943/30406193-bca25b9a-989c-11e7-8f56-38e64efa20a4.png)

**Comment by jessegreenberg** [comment]
Regarding https://github.com/phetsims/friction/issues/38#issuecomment-329330483, yes lets chat! It is cumbersome to pull styling from FocusOverlay and pass them  to options.  Maybe a common ```Path``` type in scenery-phet would be good with the default styling specified would be good.

```focusHighlight``` can also  be a ````Shape``` and if passed to the Node, FocusOverlay will apply default styling, that might work in this case.

**Comment by terracoda** [comment]
@zepumph, yes including the space with the arrows looks better to me.

**Comment by zepumph** [comment]
@terracoda @jessegreenberg In regards to the focus highligh of the chemistry book. .. . 

Here is a bit more design, but I have a few choices of the focus highlight around the chemistry book. (I marked notes about them so I don't forget, but you should just tell me which one you like most)

1.  `this.focusHighlight = Shape.bounds(bounds.withMaxY( bounds.getMaxY() - 5));`

![image](https://user-images.githubusercontent.com/6856943/30503649-874932ee-9a17-11e7-9564-9a07511bdb51.png)

2.       this.focusHighlight = Shape.bounds(bounds.erodedY(5));

![image](https://user-images.githubusercontent.com/6856943/30503679-ad4fc4ee-9a17-11e7-88d0-5ef6bac29a18.png)


3.       this.focusHighlight = Shape.bounds( bounds.eroded( 5 ) );

![image](https://user-images.githubusercontent.com/6856943/30503737-d24216d0-9a17-11e7-9524-1bf62592107c.png)

**Comment by terracoda** [comment]
Thanks for the options, @zepumph.

Hmm, there are so many lines (book edges, pages, red lines). I find the the highlight more distracting if it does align with something else. 

I'd go with number 1 for now, and see how that works.

**Comment by jessegreenberg** [comment]
@zepumph Out of the options I also prefer option 1.  What do folks think about the focus highlight going around both  books to prevent the occlusion?

**Comment by zepumph** [comment]
Interesting! I wonder if it will ever be confusing to a visual keyboard user not using a screen reader. 

Here is a version of current updates to both focus highlights. 

http://www.colorado.edu/physics/phet/dev/html/friction/1.4.0-dev.1/friction_en.html?accessibility

**Comment by jessegreenberg** [comment]
Yes, that is a good point, it what be less clear what is actually moving. What about something totally different, where the focus goes around an indication of how to move the book
![20170918_103319](https://user-images.githubusercontent.com/6396244/30553229-c0e15174-9c5d-11e7-9c0e-fe22ecf1883c.jpg)

Where the arrows are only visible when the book as keyboard focus.

**Comment by jessegreenberg** [comment]
I am worried this issue is going to get too long to be useful for documenting design decisions.  @terracoda @zepumph do you think it would  be useful to start a design document for keyboard nav in this sim? Or maybe even better would be to  add a section the existing HTML5 sim doc?

**Comment by terracoda** [comment]
@jessegreenberg, I think that's a good idea.

**Comment by zepumph** [comment]
Sounds good. Putting this issue on hold until design is solidified in a document. Unassigning for now.

**Comment by terracoda** [comment]
@zepumph, just adding one more thought until we have a design document. 

As rows of particles disappear, I think it might make sense to have the bottom edge follow the edge of yellow particles, but not shrink. This would mean the highlight would move up, so more yellow book space was part of the highlight, rather than the green particles being  included within the focus highlight.

**Comment by terracoda** [comment]
@zepumph, I created a design document called [Friction: A11y Design Main](https://docs.google.com/document/d/1XXI7SD-ASJwivsBaHgeOZmMcJFFN_s3FU48-syO-G0o/edit).

Not sure what to put in it yet, but noted the GitHub Issues.

**Comment by jessegreenberg** [comment]
Thanks @terracoda! Alright, I think this issue can be closed, we will have design documented and discussed in the google doc, and github can be used to track individual issues or changes. But if anyone disagrees, feel free to re-open the issue.
